keps/sig-windows/2258-node-service-log-viewer/README.md

```diff
@@ -0,0 +1,389 @@
+  - That have systemd / journald support.
+  - That have services that log to `/var/log/`
+- Windows worker nodes (all supported variants) that log to `C:\var\log`,
+  System and Application logs, Windows Event Logs and Event Tracing (ETW).
+
+### Non-Goals
+- Providing support for non-systemd Linux distributions.
+- Reporting logs for nodes that have config or connection issues with the
+  cluster.
+- Getting logs from services that do not use /var/log/.
+
+## Proposal
+
+### Implement client for logs endpoint viewer (OS agnostic)
+- Extend `kubectl logs` to work with node objects.
+- Implement a client for the `/var/log/` kubelet endpoint viewer.
+
+### Linux distros with systemd / journald
+Supplement the the `/var/log/` endpoint viewer on the kubelet with a thin shim
+over the `journal` directory that shells out to journalctl. Then extend
+`kubectl logs` to also work with node objects.
+
+### Linux distributions without systemd / journald
+Running the new "kubectl logs nodes" command against services on nodes that do
+not use systemd / journald should return "OS not supported". However getting
+logs from `/var/log/` should work on all systems.
+
+### Windows
+Reuse the kubelet API for querying the Linux journal for invoking the
+`Get-WinEvent` cmdlet in a PowerShell.
+
+### User Stories
+
+Consider a scenario where pods / containers are refusing to come up on certain
+nodes. As mentioned in the motivation section, troubleshooting this scenario
+involves the cluster administrator to SSH into nodes to scan the logs. Allowing
+them to use `kubectl logs` to do the same as they would to debug issues with a
+pod / container would greatly simply their debug workflow. This also opens up
+opportunities for tooling and simplifying automated log gathering. The feature
+can also be used to debug issues with Kubernetes services especially in Windows
+nodes that run as native Windows services and not as DaemonSets or Deployments.
+
+Here are some example of how a cluser administrator would use this feature:
+```
+# See what logs are available in masters in /var/log/
+kubectl logs nodes --role master --path=/
+
+# Display cron log file (/var/log/cron) from all masters
+kubectl logs nodes --role master --path=cron
+
+# Show kubelet and crio journal logs from all masters
+kubectl logs nodes --role master --path journal -s kubelet -s crio
+
+# Show kubelet log file (/var/log/kubelet/kubelet.log) from all Windows worker nodes
+kubectl logs nodes --label kubernetes.io/os=windows --path kubelet/kubelet.log
+
+# Display docker runtime WinEvent log entries from a specific Windows worker node
+kubectl logs nodes <node-name> --service docker
+
+```
+
+### Risks and Mitigations
+
+#### Large log files and events
+If the log that is attempted to be viewed is very large (GBs) there is
+potential for the node performance to be degraded. To mitigate this we can
+document that node logs should always be rotated in clusters that enable this
```

Comment:
There is no difference in node logs of large size and pod logs of large size - both can impact the node, and so this angle is strictly less risky than the current allowed pod log behavior because anyone can create pods that log large amounts of data.